### PR TITLE
fix(cairo): handle Table writer commit error in ILP

### PR DESCRIPTION
**Pull request by ideoma**
When ILP receives the data and there is a system error in commit (OOM or file limit reached) ILP code did not roll back the failing transaction which leads to SIGSEGV crash and in some cases data corruption.

The fix is:
- rollback when writer fails to commit in ILP
- disconnect ILP connections when commit error happens

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 51 / 59 (86.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|1|3|33.33%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|4|5|80.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|16|18|88.89%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cairo/CommitFailedException.java|6|6|100.00%|

File: core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java [hunk 1]
```diff
@@ -219,7 +232,19 @@ void commitIfMaxUncommittedRowsCountReached() {
...
+
+        try {
+            writer.commitWithLag();
+        } catch (Throwable th) {
+            LOG.error()
+                    .$("could not commit line protocol measurement [tableName=").$(writer.getTableName())
+                    .$(", message=").$(th.getMessage())
+                    .$(th)
+                    .I$();
+            writer.rollback();
```
Comment (ideoma) [inline core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java:244]: This is SIGSEGV crash fix

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 50 / 58 (86.21%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|15|17|88.24%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/CommitFailedException.java|6|6|100.00%|

**Review by glasstiger — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 50 / 58 (86.21%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|15|17|88.24%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/CommitFailedException.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 50 / 58 (86.21%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|2|5|40.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|15|17|88.24%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpWriterJob.java|24|27|88.89%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementEvent.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/CommitFailedException.java|6|6|100.00%|

**Pull request merged**